server/src/main/java/org/jboss/as/server/deployment/module/ModuleSpecification.java

```diff
@@ -150,18 +154,26 @@ public void addSystemDependencies(final Collection<ModuleDependency> dependencie
 
     public void addUserDependency(final ModuleDependency dependency) {
         allDependencies = null;
-        this.userDependencies.add(dependency);
+        if (!userDependenciesSet.contains(dependency.getIdentifier())) {
```

Comment:
This code can be enhanced. Just use 
this.userDependenciesSet.add(dependency.getIdentifier())
which returns boolean instead of 
if (!userDependenciesSet.contains(dependency.getIdentifier()))

Reply:
Thanks @ropalka good point! Fixed.